api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -80,6 +81,10 @@ func (mw *azureManagedMachinePoolWebhook) Default(ctx context.Context, obj runti
 		m.Spec.OSType = ptr.To(DefaultOSType)
 	}
 
+	if ptr.Deref(m.Spec.ScaleSetPriority, "") == "Spot" && m.Spec.SpotMaxPrice == nil {
```

Comment:
do we have a const in the API for "Spot"?

Reply:
yes, updated

Reply:
Yes to the SDK, not the CAPZ API